internal/index/frontmatter.go

```
@@ -0,0 +1,83 @@
+// parseFrontmatter reads the file at path, extracts the YAML block between
+// the first two `---` fences on their own lines, and unmarshals it. Missing
+// fences yield a zero-valued frontmatter and no error. Read errors and
+// malformed YAML are returned.
+func parseFrontmatter(path string) (frontmatter, error) {
+	var fm frontmatter
+
+	f, err := os.Open(path)
+	if err != nil {
+		return fm, err
+	}
+	defer f.Close()
```

golangci-lint (2.11.4)

[error] Error return value of `f.Close` is not checked

(errcheck)